### PR TITLE
adding jte block checking to pipeline config

### LINKED ISSUE #111: move JTE behavior feature flags into a root level `jte` block in the pipeline configuration
move configurations like `allow_scm_jenkinsfile` and `override_libraries` into a root level `jte` block. 

```
jte{
  allow_scm_jenkinsfile = false
  ... <-- other behavior oriented feature flags for the framework itself 
}
```

#104

### PR DESCRIPTION
**Pull request by cokieffebah**
# PR Details

Adding jte block to pipeline config
## Description

Moving allow_scm_jenkinsfile and pipeline_template into a top level 'jte' block
resolves #111 
## Types of Changes

- [x] Added Unit Testing
- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have added the appropriate label for this PR
- [x] If necessary, I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy [hunk 1]
```diff
@@ -144,7 +146,7 @@ class PipelineDecorator extends InvisibleAction {
             FileSystemWrapper fs = FileSystemWrapper.createFromJob(flowOwner)
             String repoJenkinsfile = fs.getFileContents("Jenkinsfile", "Repository Jenkinsfile", false)
             if (repoJenkinsfile){
-                Boolean allowScmJenkinsfile = pipelineConfig.containsKey("allow_scm_jenkinsfile") ? pipelineConfig.allow_scm_jenkinsfile : true
+                Boolean allowScmJenkinsfile = jteBlockWrapper.allow_scm_jenkinsfile
```
Comment (steven-terrana) [inline src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy:149]: i believe i only set this variable for readability as to not do this logic within the if statement condition. 

now that it's been abstracted into the jteBlockWrapper, we can probably do:

```groovy
if(jteBlockWrapper.allow_scm_jenkinsfile){
  return repoJenkinsfile
}
```

i think?

File: src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy [hunk 2]
```diff
@@ -189,4 +191,25 @@ class PipelineDecorator extends InvisibleAction {
...
+        /**
+         * property name should match the config keys which are snake_case/underscore-separated
+         */
+        String pipeline_template = null
+        Boolean allow_scm_jenkinsfile = true
+        JteBlockWrapper(final LinkedHashMap config){
+            List<String> fields = this.getMetaClass().properties*.getName()
+            fields.each{ field ->
+                if(config.containsKey(field)){
+                    this.setProperty(field, config[field])
```
Comment (steven-terrana) [inline src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy:205]: Should we wrap this in a try catch? 

for example, with the following pipeline configuration: 
```groovy
jte{
  pipeline_template = true
}
```

this line will likely throw a ClassCast exception bc boolean can't be converted to the String type? 

so wrapping seetProperty in try/catch to have some nicer error logging of like "jte.pipline_template must be a String" or something?

**Review by steven-terrana — CHANGES_REQUESTED** [review]

File: src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy (hunk 1 above)
Comment (cokieffebah) [reply src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy:149]: yeah, let me do that

File: src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy (hunk 2 above)
Comment (cokieffebah) [reply src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy:205]: going to add the validation.

Comment (cokieffebah) [reply src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy:204]: added validation

**Pull request merged**